components/MySQLConnection.php

```diff
@@ -47,4 +49,28 @@ public function getConnection():mysqli {
             return self::$connection;
         }
 
+        /**
+         * Returns the result of MySQLi query
+         *
+         * @param Query $query
+         * @return mysqli_result|bool
+         */
+        public function query(Query $query) {
+            return self::$connection->query($query->getQueryString());
+        }
+
+        /**
+         * Returns all matched rows from DB Table
+         *
+         * @param Query $query
+         * @param int $resultType
+         * @return array
+         */
+        public function fetchAll(Query $query, int $resultType = MYSQLI_ASSOC):array {
+            $result  = $this->query($query);
+            $fetched = $result->fetch_all($resultType);
+
+            return $fetched;
+        }
```

Comment:
Reduces code repetition as the sequence of 'result; fetch_all' had a lot of occurances.